jmbitcoin/jmbitcoin/secp256k1_transaction.py

```diff
@@ -98,18 +108,18 @@ def read_flag_byte(val):
     for i in range(ins):
         obj["ins"].append({
             "outpoint": {
-                "hash": read_bytes(32)[::-1],
+                "hash": hex_string(ser_read(tx, 32)[::-1], hexout),
                 "index": read_as_int(4)
             },
             #TODO this will probably crap out on null for segwit
-            "script": read_var_string(),
+            "script": hex_string(read_var_string(), hexout),
             "sequence": read_as_int(4)
         })
     outs = read_var_int()
     for i in range(outs):
         obj["outs"].append({
             "value": read_as_int(8),
-            "script": read_var_string()
+            "script": hex_string(read_var_string(), hexout)
         })
     #segwit flag is only set if at least one txinwitness exists,
     #in other words it would have to be at least partially signed;
```

Comment:
Just to note my comment here a little lower down "with the items in the witness space separated" is wrong of course, it's a list (not sure what that was about).